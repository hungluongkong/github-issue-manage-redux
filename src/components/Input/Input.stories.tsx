import React from 'react';
import { storiesOf } from '@storybook/react';
import Input from '.';
import { PreviewArea, Code, DetailTable } from '../commonStyled';

// Detail of props
const propsDetail: JSX.Element = DetailTable({
  data: [
    { name: 'defaultValue', type: 'String' },
    { name: 'placeholder', type: 'String' },
    { name: 'isRequire', type: 'boolean', default: 'false' },
    { name: 'textRef', type: 'React.RefObject<HTMLInputElement>' },
  ],
});

storiesOf('Input', module)
  .add('Default', () => (
    <>
      <h3>Default Input</h3>
      <PreviewArea>
        <Input defaultValue="Text" placeholder="Input here" />
      </PreviewArea>
      <Code>
        {`
          <Input defaultValue="Text" placeholder="Input here" />
        `}
      </Code>
      {propsDetail}
    </>
  ));
