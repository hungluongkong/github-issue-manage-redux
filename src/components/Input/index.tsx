import React from 'react';
import styled from 'styled-components';

import * as color from '../../theme/color';
import { InputProps } from '../../types';

const InputStyled = styled.input`
  padding: 5px 12px;
  background-color: ${color.INPUT_BG_COLOR};
  line-height: 20px;
  border: 1px solid ${color.INPUT_BORDER_COLOR};
  border-radius: 6px;
  box-sizing: border-box;
  flex-grow: 1;
  width: 100%;

  :focus {
    box-shadow: 1px 1px 1px 1px ${color.SHADOW_BORDER};
  }
`;

const Input = ({
  defaultValue = '',
  isRequire = false,
  placeholder = '',
  textRef,
  onChange = () => {},
}: InputProps): JSX.Element => (
  <InputStyled
    placeholder={placeholder}
    defaultValue={defaultValue}
    ref={textRef as React.RefObject<HTMLInputElement>}
    onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
      onChange(e.target.value)
    }
    required={isRequire}
  />
);

export default Input;
