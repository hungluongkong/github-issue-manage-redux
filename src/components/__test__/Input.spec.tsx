import React, { useRef } from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Input from '../Input';

describe('Unit test for Input component', () => {
  it('snapshot', () => {
    const input = renderer.create(<Input />);
    expect(input).toMatchSnapshot();
  });

  it('render Default value', () => {
    const defaultValue = 'Test text';
    const placeholderText = 'Place holder text';
    const input = shallow(
      <Input defaultValue={defaultValue} placeholder={placeholderText} />,
    );
    expect(input.prop('defaultValue')).toEqual(defaultValue);
    expect(input.prop('placeholder')).toEqual(placeholderText);
  });

  it('simulate on change', () => {
    const defaultValue = 'Default text';
    const newValue = 'Changed text';
    const onChange = jest.fn();

    const input = mount(
      <Input defaultValue={defaultValue} onChange={onChange} />,
    );

    input.simulate('change', { target: { value: newValue } });

    expect(onChange).toBeCalledWith(newValue);
  });
});
