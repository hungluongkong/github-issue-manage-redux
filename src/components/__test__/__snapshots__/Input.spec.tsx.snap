// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for Input component snapshot 1`] = `
<input
  className="sc-AxjAm knKJQz"
  defaultValue=""
  onChange={[Function]}
  placeholder=""
  required={false}
/>
`;
